### ISSUE TITLE
Docs: Scroll Restoration broken

### ISSUE THREAD

**Issue by jantimon** [vercel/next.js#70148]
### What is the documentation issue?

When navigating through the Next.js documentation, there's an issue with the back button functionality that might be a bug in the APP router

## Steps to Reproduce

1. Go to https://nextjs.org/docs/app/building-your-application/caching#request-memoization
2. Click on "Fetch API" in the content
3. Click on "Fetch API Reference"
4. Use the browser's back button

https://github.com/user-attachments/assets/0ef14151-a9d6-47d1-9f3a-31306152cec0

### Is there any context that might help us understand?

This video was recorded in Brave

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/caching#request-memoization

**Comment by samcx** [comment]
@jantimon Thanks for sharing! We are taking a look.

I see that if you click the back button twice, the scroll restoration goes to the correct page.

**Comment by pakeerathanchandrabalael** [comment]
It seems that scroll restoration in the App Router is not working as expected. When navigating to a page using the Link component and then returning with the browser's back button, the scroll position is not preserved. Any update on this?

**Comment by isaackwok** [comment]
facing the same problem in my own Next.js project (15.1.6) with app router.

**Comment by samcx** [comment]
> It seems that scroll restoration in the App Router is not working as expected. When navigating to a page using the Link component and then returning with the browser's back button, the scroll position is not preserved. Any update on this?

@jantimon @pakeerathanchandrabalael @isaackwok I don't think this is a scrolling bug in App Router, but some weird element causing it to scroll down it.

Will be taking a look!

**Comment by iTzRusso10** [comment]
I have the same problem in my Next 15 project with app routing, when you navigate between routes with <Link /> scroll restoration works randomly while other times it doesn't work at all, SPECIALLY ON MOBILE

**Comment by V-iktor** [comment]
Same thing here.

I have a list of products with a skeleton loader.

If I scroll to bottom of a category, visit a product, and press back, I am back to the top of the category page.

```
app/
├── product/[slug]/
│  └── page.tsx
└── category/
   └── [...categories]/
      ├── loading.tsx
      └── page.tsx
```

` "next": "15.1.6"`

**Comment by l-you** [comment]
@jantimon @iTzRusso10 you can use https://github.com/RevoTale/next-scroll-restorer
I gave up messaging Next.js team about that issues. So I created well-tested scroll restoration solution as a separate client component.

**Comment by phil-sola** [comment]
Facing exactly the same issue in my nextjs application using 15.2.3 and the app router.

Desperately need a fix for this! 

Just been logging some weird findings in my app:

🔗 CLICK - Leaving page at scrollY: 414 RouteChangeEffects.tsx:47:33

⬅️ POPSTATE - Expected scrollY: 414 RouteChangeEffects.tsx:66:29

⬅️ POPSTATE - Actual scrollY after 100ms: 815.5 RouteChangeEffects.tsx:70:37

⬅️ POPSTATE - Actual scrollY after 500ms: 815.5 RouteChangeEffects.tsx:75:37

**Comment by phil-sola** [comment]
> > It seems that scroll restoration in the App Router is not working as expected. When navigating to a page using the Link component and then returning with the browser's back button, the scroll position is not preserved. Any update on this?
> 
> [@jantimon](https://github.com/jantimon) [@pakeerathanchandrabalael](https://github.com/pakeerathanchandrabalael) [@isaackwok](https://github.com/isaackwok) I don't think this is a scrolling bug in App Router, but some weird element causing it to scroll down it.
> 
> Will be taking a look!

@samcx Did you ever get anywhere with this and uncovering the cause? Seems a lot of folks still having issues with this in the app router.

**Comment by marcoshernanz** [comment]
Thanks for the report. I retested the original flow on the versioned Next.js 14 docs, and scroll restoration works correctly. The first Back returns to `#fetch`, and the second returns to `#request-memoization`, since the in-page hash link creates its own history entry.

The broader failure reported later, where revalidation or `router.refresh()` invalidates the Router Cache before Back navigation, is the same limitation covered in #70716 and recently deduplicated in #86528. Since the original docs flow is resolved and the remaining case is already covered there, I’m closing this.
